api/wallet.go

```diff
@@ -367,42 +368,39 @@ func (api *API) walletSeedsHandler(w http.ResponseWriter, req *http.Request, _ h
 
 // walletSiacoinsHandler handles API calls to /wallet/siacoins.
 func (api *API) walletSiacoinsHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
-	amountStrs := strings.Split(req.FormValue("amount"), ",")
-	destStrs := strings.Split(req.FormValue("destination"), ",")
-	amounts := make([]types.Currency, len(amountStrs))
-	dests := make([]types.UnlockHash, len(destStrs))
-	for i, amountStr := range amountStrs {
-		var ok bool
-		amounts[i], ok = scanAmount(amountStr)
-		if !ok {
-			WriteError(w, Error{"could not read amount from POST call to /wallet/siacoins"}, http.StatusBadRequest)
+	var txns []types.Transaction
+	if req.FormValue("outputs") != "" {
+		// multiple amounts + destinations
+		var outputs []types.SiacoinOutput
+		err := json.Unmarshal([]byte(req.FormValue("outputs")), &outputs)
```

Comment:
need to verify before parsing that 'amount' and 'destination' are both empty - don't want a user filling out both and getting confused.